src/pydata_sphinx_theme/utils.py

```diff
@@ -58,3 +60,79 @@ def maybe_warn(app: Sphinx, msg, *args, **kwargs):
     should_warn = theme_options.get("surface_warnings", False)
     if should_warn:
         SPHINX_LOGGER.warning(msg, *args, **kwargs)
+
+
+def set_secondary_sidebar_items(
+    app: Sphinx, pagename: str, templatename: str, context, doctree
+) -> None:
+    """Set the secondary sidebar items to render for the given pagename."""
+    if "theme_secondary_sidebar_items" in context:
+        templates = context["theme_secondary_sidebar_items"]
+        if isinstance(templates, dict):
+            templates = _get_matching_sidebar_items(pagename, templates)
+
+        context["secondary_sidebar_items"] = _update_and_remove_templates(
+            app,
+            context,
+            templates,
+            "theme_secondary_sidebar_items",
+        )
+
+
+def _update_and_remove_templates(
```

Comment:
it looks like a lot of this function was copy-pasted from `update_and_remove_templates` in our `__init__.py`.  Is there a way to refactor so we don't have duplicated code?  (e.g., a shared helper function that they both call?)

Reply:
For sure!